pinot-plugins/pinot-input-format/pinot-orc/src/main/java/org/apache/pinot/plugin/inputformat/orc/ORCRecordReader.java

```diff
@@ -106,6 +111,18 @@ public void init(File dataFile, @Nullable Set<String> fieldsToRead, @Nullable Re
     _nextRowId = 0;
   }
 
+  private File unpackIfRequired(File dataFile) throws IOException {
```

Comment:
Let's move this to `RecordReaderUtils` class

Reply:
done!
I decided to add `extension` as a function parameter explicitly, as you've mentioned above the archive will not necessarily have `.gz` suffix.